File: digit_ops/digit_ops.py
def calculate_digit_ops(num):
    """
    Вычисляет сумму и произведение цифр числа, а также модуль их разности.
    
    Аргументы:
        num (int): Натуральное число для обработки
    
    Возвращает:
        None: Выводит результат в формате "sum product abs_diff"
    """
    sum_digits = 0
    product = 1
    n = num
    
    while n > 0:
        digit = n % 10
        sum_digits += digit
        product *= digit
        n = n // 10
    
    print(f"{sum_digits} {product} {abs(sum_digits - product)}")

N = int(input())
calculate_digit_ops(N)

File: digit_ops/digit_ops.rs
use std::io;

/// Вычисляет сумму и произведение цифр числа, а также модуль их разности.
/// 
/// # Аргументы
/// * `num` - Натуральное число для обработки

fn calculate_digit_ops(num: i32) {
    let mut n = num;
    let mut sum = 0;
    let mut product = 1;
    
    while n > 0 {
        let digit = n % 10;
        sum += digit;
        product *= digit;
        n /= 10;
    }
    
    println!("{} {} {}", sum, product, (sum - product).abs());
}

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let N: i32 = input.trim().parse().unwrap();
    calculate_digit_ops(N);
}